File: ConsequencesClientExample/ConsequencesClientExampleTests/Websocket/SocketClientTests.cs
using ConsequencesClientExample.Websocket;

namespace ConsequencesClientExampleTests.Websocket
{
    internal class SocketClientTests
    {
        [Test]
        public void Receive_WhenMessageListManuallyFilled_ReturnsMessage()
        {
            // Arrange
            SocketClient socketClient = new SocketClient();
            socketClient._responseList.Add("Test");

            // Act
            var response = socketClient.Receive();

            // Assert
            Assert.That(response, Is.EqualTo("Test"));
        }
    }
}

File: Starter Projects/Java/ConsequencesStarterClient/src/test/java/SocketClientTests.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class SocketClientTests {

    @Test
    void demoTestMethod() {
        // Arrange
        SocketClient socketClient = new SocketClient("ws://51.141.52.52:1234");

        // Act
        OutboundMessage message = new OutboundMessage();
        message.Hello = "Hello";
        socketClient.Send(message);
        var response = socketClient.Receive();

        // Assert
        assertEquals("Welcome to Consequences, to get started send your name and room code.", response.Message);
    }
}
